### ISSUE TITLE
bug: programs.rbw broken

### ISSUE THREAD

**Issue by srhb** [nix-community/home-manager#2476]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue description

rbw has begun trying to mutate its own config file during normal use. This obviously breaks.

```
rbw get: failed to save config to /home/sarah/.config/rbw/config.json: Read-only file system (os error 30)
```

### Maintainer CC

@ambroisie 

### System information

```markdown
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.79, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.5pre20211007_844dd90`
 - channels(root): `""`
 - channels(sarah): `""`
 - nixpkgs: `/home/sarah/src/nixpkgs`
```
```

**Comment by rycee** [comment]
That's kind of a bummer. We don't really have a good response to such program behavior in HM. I guess we could make the module install the file using an activation block but that's kind of ugly and would only apply during activation. I hope this can be fixed upstream instead 🙂

**Comment by srhb** [comment]
Agreed! Just tracking here, hope that's okay. :) I'm sure we can work something out with upstream.

**Comment by akho** [comment]
For now, setting `device_id` works and prevents rbw from misbehaving

```
  programs.rbw.settings.device_id = "<put uuid here>";
```

Not a good long-term solution at all, though.

**Comment by ivankovnatsky** [comment]
This is related to this:

* https://github.com/doy/rbw/issues/71
* https://github.com/bitwarden/cli/issues/383

> Agreed! Just tracking here, hope that's okay. :) I'm sure we can work something out with upstream.

so the upstream here being probably bitwarden not rbw.

While `device_id` is obviously will be different for a different machine this makes a bit harder to keep a declarative configuration of rbw.

**Comment by berbiche** [comment]
Should we mark the module as broken somehow?
If "`settings` is a set and it does not contain a `device_id`" then we print a warning?

**Comment by ivankovnatsky** [comment]
Nah, it should be fixed in 1.4.2: https://github.com/doy/rbw/commit/42badc1f3281961bbfadffffb64de913f5b6084f.

**Comment by berbiche** [comment]
Alright, this can be marked as resolved then!
